source/client/process_bootstrap.cc

```diff
@@ -14,28 +14,130 @@
+  if (options.transportSocket().has_value()) {
+    return options.transportSocket().value();
+  }
+
+  TransportSocket transport_socket;
+  transport_socket.set_name("envoy.transport_sockets.tls");
+
+  UpstreamTlsContext upstream_tls_context = options.tlsContext();
+  const std::string sni_host = Client::SniUtility::computeSniHost(uris, options.requestHeaders(),
+                                                                  options.upstreamProtocol());
+  if (!sni_host.empty()) {
+    *upstream_tls_context.mutable_sni() = sni_host;
+  }
+
+  CommonTlsContext* common_tls_context = upstream_tls_context.mutable_common_tls_context();
+  if (options.upstreamProtocol() == Envoy::Http::Protocol::Http2) {
+    common_tls_context->add_alpn_protocols("h2");
+
+  } else if (options.upstreamProtocol() == Envoy::Http::Protocol::Http3) {
+    return absl::UnimplementedError("HTTP/3 Quic support isn't implemented yet.");
+
+  } else {
+    common_tls_context->add_alpn_protocols("http/1.1");
+  }
+
+  transport_socket.mutable_typed_config()->PackFrom(upstream_tls_context);
+  return transport_socket;
+}
+
+// Creates circuit breakers configuration based on the specified options.
+CircuitBreakers createCircuitBreakers(const Client::Options& options) {
+  CircuitBreakers circuit_breakers;
+  CircuitBreakers::Thresholds* thresholds = circuit_breakers.add_thresholds();
+
+  // We do not support any retrying.
+  thresholds->mutable_max_retries()->set_value(0);
+  thresholds->mutable_max_connections()->set_value(options.connections());
+
+  // We specialize on 0 below, as that is not supported natively. The benchmark client will track
+  // in flight work and avoid creating pending requests in this case.
+  thresholds->mutable_max_pending_requests()->set_value(
+      options.maxPendingRequests() == 0 ? 1 : options.maxPendingRequests());
+  thresholds->mutable_max_requests()->set_value(options.maxActiveRequests());
+
+  return circuit_breakers;
+}
+
+// Creates a cluster used by Nighthawk to upstream requests to the uris by the specified worker
+// number.
+Cluster createNighthawkClusterForWorker(const Client::Options& options,
+                                        const std::vector<UriPtr>& uris, int worker_number) {
+  Cluster cluster;
+
+  cluster.set_name(fmt::format("{}", worker_number));
+  cluster.mutable_connect_timeout()->set_seconds(options.timeout().count());
+  cluster.mutable_max_requests_per_connection()->set_value(options.maxRequestsPerConnection());
+
+  if (options.upstreamProtocol() == Envoy::Http::Protocol::Http2) {
+    auto* cluster_http2_protocol_options = cluster.mutable_http2_protocol_options();
+    cluster_http2_protocol_options->mutable_max_concurrent_streams()->set_value(
+        options.maxConcurrentStreams());
+  }
+
+  *cluster.mutable_circuit_breakers() = createCircuitBreakers(options);
+
+  cluster.set_type(Cluster::DiscoveryType::Cluster_DiscoveryType_STATIC);
```

Comment:
Since enums don't create a namespace, the correct expression is even shorter: `Cluster::STATIC`.

Changed both locations in 94badbb2c2773ae8277c7daf2fd08060e98fb2f7.